processor/memorylimiterprocessor/factory.go

```diff
@@ -122,7 +122,7 @@ func (f *factory) getMemoryLimiter(set component.ProcessorCreateSettings, cfg co
 	f.lock.Lock()
 	defer f.lock.Unlock()
 
-	if memLimiter, ok := f.memoryLimiters[cfg.ID()]; ok {
+	if memLimiter, ok := f.memoryLimiters[cfg]; ok {
```

Comment:
Why does it need to be changed? just using another wrapping struct as the key, right? cannot see any reasons for that

Reply:
https://github.com/open-telemetry/opentelemetry-collector/issues/4714